biz.aQute.launchpad/src/aQute/launchpad/StandaloneLaunchpadBuilder.java

```diff
@@ -0,0 +1,290 @@
+package aQute.launchpad;
+
+import static aQute.bnd.osgi.Constants.EXPORT_CONTENTS;
+import static aQute.bnd.osgi.Constants.SERVICE_COMPONENT;
+import static aQute.lib.exceptions.Exceptions.unchecked;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.jar.Manifest;
+import java.util.stream.Collectors;
+
+import aQute.bnd.build.Container;
+import aQute.bnd.build.Project;
+import aQute.bnd.build.Run;
+import aQute.bnd.header.Attrs;
+import aQute.bnd.header.Parameters;
+import aQute.bnd.osgi.About;
+import aQute.bnd.osgi.Builder;
+import aQute.bnd.osgi.Descriptors.PackageRef;
+import aQute.bnd.osgi.Jar;
+import aQute.bnd.osgi.Processor;
+import aQute.bnd.service.Strategy;
+import aQute.bnd.service.remoteworkspace.RemoteWorkspace;
+import aQute.bnd.service.specifications.BuilderSpecification;
+import aQute.bnd.service.specifications.RunSpecification;
+import aQute.bnd.version.Version;
+import aQute.bnd.version.VersionRange;
+import aQute.lib.utf8properties.UTF8Properties;
+import aQute.libg.glob.Glob;
+
+public class StandaloneLaunchpadBuilder extends AbstractLaunchpadBuilder<StandaloneLaunchpadBuilder> {
```

Comment:
We should not have a separate builder as far as I can see. That makes the source code incompatible.

The only problem with the current approach is that the client searches for the workspace using the working directory to find the workspace cache directory. This all is only needed to find the port the 'current' workspace is registered on.

In the maven case, we create the workspace ourselves. We could this easily find out the port of the workspace and register this as a system property when launching the test process.

Reply:
Each execution instance in maven or gradle.builder has a unique workspace so there is nothing to link to. Also, I am remiss to handle this as a remote process.